### DISCUSSION TITLE
Template Changes

### DISCUSSION THREAD

**Discussion by chicagocomputerclasses** [fireactjs/core#2] [Q&A]
As far as I understand, the only way to modify the HTML structure for core pages (login,logout,navmenu,etc) is to modify the files in the core library. Is that correct or did I miss something important?

**Comment by chaoming** [comment]
For customizing the HTML structure, I suggest you create your own `PublicTemplate` component to wrap the existing `SignIn` and other functional components.

```
<Route element={<PublicTemplate />}>
```

Replace this one.

**Reply by chicagocomputerclasses** [reply]
Thanks for the quick response.

Using this same approach for AppTemplate.js, what would I import instead of import { FireactContext } from "../Fireact" ?

**Reply by chaoming** [reply]
No worries.

`FireactContext` contains the configurations of your project, so you should still use it as it is in the `AppTemplate` example here: https://github.com/fireactjs/core/blob/main/src/lib/components/templates/AppTemplate.js

For details of the HTML structure in the `AppTemplate` component, please see the document here: https://fireactjs.com/docs/core-package/components/apptemplate/

**Reply by chaoming** [reply]
@chicagocomputerclasses sorry, I misunderstood your question. To import `FireactContext`, you will need to import it from the package as 
```
import {FireactContext} from '@fireactjs/core';
```

**Reply by chicagocomputerclasses** [reply]
That makes sense. Thank You!
